### ISSUE TITLE
Serializations fails with JaxbAnnotation Introspector and intermediary objects

### ISSUE THREAD

**Issue by jej2003** [FasterXML/jackson-module-jaxb-annotations#56]
(moved from https://github.com/FasterXML/jackson-databind/issues/1143)

------

I have some JAXB annotated classes that I'm using the `JaxbAnnotationIntrospector` with that generally speaking works well, but when I have an intermediary object being created that then has a simple string value I have issues converting to JSON.  A sample snippet demonstrating the problem is shown below.

Should this work or am I misusing something?

The exception is

```
com.fasterxml.jackson.databind.JsonMappingException: No serializer found for class java.lang.String and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS) ) (through reference chain: jackson.DTO["map"]->jackson.Intermediary["entry"]->java.util.ArrayList[0]->jackson.IntermediaryEntry["value"])
```

```java
public class TestClass {

    @Test
    public void test() throws Exception {
        DTO d = new DTO();
        d.map = new HashMap<>();
        d.map.put("key", "value");

        ObjectMapper mapper = new ObjectMapper();
        mapper.setAnnotationIntrospector(new JaxbAnnotationIntrospector(mapper.getTypeFactory()));

        mapper.writeValueAsString(d);
    }



    static public class MapAdapter extends XmlAdapter<Intermediary, Map<String, Object>> {

        @Override
        public Intermediary marshal(Map<String, Object> v) throws Exception {
            if(v == null) {
                return null;
            }
            Intermediary i = new Intermediary();
            for(Entry<String, Object> entry : v.entrySet()) {
                i.add(new IntermediaryEntry(entry));
            }

            return i;
        }

        @Override
        public Map<String, Object> unmarshal(Intermediary v) throws Exception {
            Map<String, Object> obj = new HashMap<>();
            if(v != null &&  v.entries != null) {

                for (IntermediaryEntry e : v.entries) {
                    obj.put(e.key, e.value);
                }
            }
            return Collections.unmodifiableMap(obj);
        }

    }

    @XmlAccessorType(XmlAccessType.FIELD)
    static public class Intermediary {
        @XmlElement(name="entry")
        List<IntermediaryEntry> entries;
        public void add(IntermediaryEntry e) {
            if(entries == null) {
                entries = new ArrayList<>();
            }
            entries.add(e);
        }
    }

    @XmlAccessorType(XmlAccessType.FIELD)
    static public class IntermediaryEntry {
        String key;
        @XmlJavaTypeAdapter(PrimitiveStringOrDateAdapter.class)
        Object value;
        public IntermediaryEntry() {

        }
        public IntermediaryEntry(Entry<String, Object> entry) {
            this.key = entry.getKey();
            this.value = entry.getValue();
        }
    }

    static public class PrimitiveStringOrDateAdapter extends XmlAdapter<Object, Object> {
        @Override
        public Object unmarshal(Object v) throws Exception {
            if(v instanceof XMLGregorianCalendar) {
                return  ((XMLGregorianCalendar)v).toGregorianCalendar().getTime();
            }
            return v;
        }

        @Override
        public Object marshal(Object v) throws Exception {
            return v;
        }
    }

    public class DTO {
        @XmlElement
        @XmlJavaTypeAdapter(MapAdapter.class)
        Map<String, Object> map;
    }
}
```

**Comment by cowtowncoder** [comment]
Hmmh. I can not say off-hand, since example is quite complicated. :-)

My first guess would have been that serialization of root-level `Map` is problematic, due to type erasure (I recommend not using generic types as root value these days -- too many issues get reported). But that may not be the problem as adapters do seem to get invoked.

Another possible lead is that Jackson can not serialize plain `Object` type. But when dynamic typing (that is, use of runtime type, not just declared) is enabled, it can detect the actual type... so that too should actually work.

At this point what would be useful would be:

1. Stack trace (to show where exception gets thrown)
2. Version used -- reproduction on 2.7.2 (released just a day ago) would be great, although behavior wrt JAXB module hasn't changed for a while. But 2.7 does have type handling improvements.

**Comment by jej2003** [comment]
It looks like this was fixed in 2.5.3 (was running 2.5.0).  I've since taken another approach as the extra object structures were only being created for some JAXB related processing.  I've instead added the Jackson annotations and it's much simpler of an approach vs trying to wrestle with this.  Thanks for the quick response.

**Comment by cowtowncoder** [comment]
@jej2003 Ok good! Thank you for verifying this.
